### ISSUE TITLE
Potentially incorrect allocation to fine roots in PRTAllometricCarbonMod.F90?

### ISSUE THREAD

**Issue by mpaiao** [NGEET/fates#784]
I was looking at this code and it seems to me that allocation to fine roots would be underestimated:

https://github.com/NGEET/fates/blob/70a3e133244877f325652f2cf40e637d785b3aa7/parteh/PRTAllometricCarbonMod.F90#L543-L555

Carbon balance is subtracted after finding leaf_c_flux, and then scaled by fnrt_c_demand/total_c_demand, so the relative demand of fine root demand is applied in a lower carbon balance value. Shouldn't the code be something like the one below?
```fortran
       leaf_c_flux    = min(leaf_c_demand, &
                        carbon_balance*(leaf_c_demand/total_c_demand))
       fnrt_c_flux    = min(fnrt_c_demand, &
                            carbon_balance*(fnrt_c_demand/total_c_demand))
       leaf_c(iexp_leaf) = leaf_c(iexp_leaf) + leaf_c_flux
       fnrt_c         = fnrt_c + fnrt_c_flux
       carbon_balance = carbon_balance - leaf_c_flux - fnrt_c_flux
```

**Comment by mpaiao** [comment]
And I think the same rationale applies to the block a few lines above.

https://github.com/NGEET/fates/blob/70a3e133244877f325652f2cf40e637d785b3aa7/parteh/PRTAllometricCarbonMod.F90#L485-L506

**Comment by rgknox** [comment]
I agree @mpaiao

**Comment by rgknox** [comment]
yeah, we should not be updating the carbon balance until both pools have calculated their fluxes

**Comment by jkshuman** [comment]
I also agree with you @mpaiao good catch.

**Comment by rgknox** [comment]
I like simple, seems fine to have the alternative.
In the N & P enabled allocation scheme, we allow the user to define the order in which replace tissues.  We could potentially merge our ideas and create a c-only version:
https://github.com/NGEET/fates/blob/master/parteh/PRTAllometricCNPMod.F90#L615

**Comment by glemieux** [comment]
Closing per https://github.com/NGEET/fates/pull/863
